DesignSafe-CI/portal
# task/des-2657--job-history-listing

## Overview: ##
Fill in the Jobs Status table to resemble the [mock-up](https://xd.adobe.com/view/bab9a62d-3f2a-45f8-be0f-5680c474d9c1-efea/). 
## PR Status: ##


## Related Jira tickets: ##

* [DES-2657](https://tacc-main.atlassian.net/browse/DES-2657)

## Summary of Changes: ##
**Columns, and where the data is coming from:** 
  Job Name --> `record.name`
  Application --> `appId`
  Job Status --> `status`, uses getStatusText
  Nodes --> `nodeCount`
  Cores --> `coresPerNode`
  Time Submitted - Finished --> `record.created`, `record.ended`

**Buttons**
View Output --> appears when `status === ('FAILED' || 'FINISHED')`; links to `{data/browser/${record.archiveSystemId}${record.archiveSystemDir}}`
Output Pending --> disabled button that appears when `status === ('PENDING' || 'PROCESSING_INPUTS' || 'STAGING_INPUTS' || 'STAGING_JOB' || 'SUBMITTING_JOB' || 'QUEUED')`
Open --> button should only appear for Interactive apps. [NOT COMPLETE, currently just a placeholder]
View Details --> link that will open to the Jobs Modal


## Testing Steps: ##
1. Click 'Job Status', verify table loads. 
2. Verify table content is correct.
3. Check that the View Output button works.

## UI Photos:
<img width="1316" alt="Screen Shot 2024-05-16 at 4 54 23 PM" src="https://github.com/DesignSafe-CI/portal/assets/35277477/25d9cb8d-25b5-467d-8bc9-765cefb5e9df">

## Notes: ##
Open button needs to be connected. 
The table height is small so you end up having to scroll to see all rows.
